File: lib/model/geometry_change.dart
part of graph.model;

class GeometryChange extends AtomicGraphModelChange {

  Object cell;

  Geometry geometry, previous;

  GeometryChange([GraphModel model = null, this.cell = null, this.geometry = null]) : super(model) {
    this.previous = this.geometry;
  }

  void setCell(Object value) {
    cell = value;
  }

  /**
   * Returns the cell.
   */
  Object getCell() {
    return cell;
  }

  void setGeometry(Geometry value) {
    geometry = value;
  }

  /**
   * Returns the geometry.
   */
  Geometry getGeometry() {
    return geometry;
  }

  void setPrevious(Geometry value) {
    previous = value;
  }

  /**
   * Returns the previous.
   */
  Geometry getPrevious() {
    return previous;
  }

  /**
   * Changes the root of the model.
   */
  void execute() {
    geometry = previous;
    previous = (model as GraphModel)._geometryForCellChanged(cell, previous);
  }

}

File: src/graph/model/GeometryChange.java
package graph.model;

//import graph.model.IGraphModel.AtomicGraphModelChange;

public class GeometryChange extends AtomicGraphModelChange
{

	/**
	 *
	 */
	protected Object cell;

	/**
	 * 
	 */
	protected Geometry geometry, previous;

	/**
	 * 
	 */
	public GeometryChange()
	{
		this(null, null, null);
	}

	/**
	 * 
	 */
	public GeometryChange(GraphModel model, Object cell,
			Geometry geometry)
	{
		super(model);
		this.cell = cell;
		this.geometry = geometry;
		this.previous = this.geometry;
	}

	/**
	 * 
	 */
	public void setCell(Object value)
	{
		cell = value;
	}

	/**
	 * @return the cell
	 */
	public Object getCell()
	{
		return cell;
	}

	/**
	 *
	 */
	public void setGeometry(Geometry value)
	{
		geometry = value;
	}

	/**
	 * @return the geometry
	 */
	public Geometry getGeometry()
	{
		return geometry;
	}

	/**
	 *
	 */
	public void setPrevious(Geometry value)
	{
		previous = value;
	}

	/**
	 * @return the previous
	 */
	public Geometry getPrevious()
	{
		return previous;
	}

	/**
	 * Changes the root of the model.
	 */
	public void execute()
	{
		geometry = previous;
		previous = ((GraphModel) model)._geometryForCellChanged(cell,
				previous);
	}

}